src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -14037,6 +14037,9 @@ partial class C1
             var compilation = CreateCompilation(new[] { source }, options: WithNonNullTypesTrue());
 
             compilation.VerifyDiagnostics(
+                // (10,18): error CS0761: Partial method declarations of 'C1.M1<T>(T?, T[]?, Action<T?>, Action<T?[]?>?[]?)' have inconsistent type parameter constraints
+                //     partial void M1<T>(T? x, T[]? y, System.Action<T?> z, System.Action<T?[]?>?[]? u) where T : class
+                Diagnostic(ErrorCode.ERR_PartialMethodInconsistentConstraints, "M1").WithArguments("C1.M1<T>(T?, T[]?, System.Action<T?>, System.Action<T?[]?>?[]?)").WithLocation(10, 18),
```

Comment:
>Diagnostic [](start = 16, length = 10)

This looks wrong to me

Reply:
If they differ by nullable annotations only, this should be a nullability warning, not an error.

---
In reply to: [272738601](https://github.com/dotnet/roslyn/pull/34797#discussion_r272738601) [](ancestors = 272738601)

Reply:
> If they differ by nullable annotations only, this should be a nullability warning, not an error.

At the moment, nullable mismatch causes an error. We do have an issue [https://github.com/dotnet/roslyn/issues/30229](https://github.com/dotnet/roslyn/issues/30229) to finalize the design and behavior around nullability mismatch in partials. For the purpose of this PR, the error is good enough to verify an expectation that there is a mismatch, even though the error might be converted to a warning, or go away completely in the future.


---
In reply to: [272742716](https://github.com/dotnet/roslyn/pull/34797#discussion_r272742716) [](ancestors = 272742716,272738601)

Reply:
OK, please add a comment.

---
In reply to: [272766054](https://github.com/dotnet/roslyn/pull/34797#discussion_r272766054) [](ancestors = 272766054,272742716,272738601)